### PR TITLE
Lua5.3 create shared library

### PR DESCRIPTION
**Pull request by cwluxul**
I'm submitting this pull request so that the lua5.3 package will build a shared library which can be linked by other software.  I followed the example from the regular lua package on the Makefile, and recreated the shared library patch (020-shared_liblua.patch) from the regular lua package.

I also found offset warnings with the 100-no_readline.patch and have fixed those offsets and included that commit in this pull request also.  

I built and tested on BCM5301X and the interpreter and shared object are linked and working correctly.

### CODE REVIEW

File: package/utils/lua5.3/Makefile [hunk 1]
```diff
@@ -40,8 +40,22 @@ define Package/lua5.3/Default/description
   programming, data-driven programming, and data description.
 endef
 
+define Package/liblua5.3
+$(call Package/lua53/Default)
+  SUBMENU:=
+  SECTION:=libs
+  CATEGORY:=Libraries
+  TITLE+= (libraries)
+endef
```
Comment (rmilecki) [inline package/utils/lua5.3/Makefile]: Please keep `ABI_VERSION` there just like in the `package/utils/lua/Makefile`. It's required to trigger recompilation of all dependant software on every library ABI change.

File: package/utils/lua5.3/Makefile [hunk 2]
```diff
@@ -108,7 +123,14 @@ define Build/InstallDev
 	$(INSTALL_DIR) $(1)/usr/include/lua5.3 $(1)/usr/lib
 	$(CP) $(PKG_INSTALL_DIR)/usr/include/lua5.3/lua{,lib,conf}.h $(1)/usr/include/lua5.3/
 	$(CP) $(PKG_INSTALL_DIR)/usr/include/lua5.3/lauxlib.h $(1)/usr/include/lua5.3/
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/liblua5.3.a $(1)/usr/lib/
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/liblua5.3.{a,so*} $(1)/usr/lib/
+	$(LN) liblua5.3.so.$(PKG_VERSION) $(1)/usr/lib/liblualib5.3.so
```
Comment (rmilecki) [inline package/utils/lua5.3/Makefile]: This results in `liblualib5.3.so` being a symlink to `liblua5.3.so.5.3.5`. The later one doesn't exist.

Comment (cwluxul) [reply package/utils/lua5.3/Makefile]: @rmilecki - I've resolved this by linking the actual library version.

File: package/utils/lua5.3/Makefile (hunk 1 above)
Comment (cwluxul) [reply package/utils/lua5.3/Makefile]: @rmilecki  - I've added this ABI_VERSION

File: package/utils/lua5.3/Makefile (hunk 2 above)
Comment (cwluxul) [reply package/utils/lua5.3/Makefile]: @rmilecki - One other thing to note, I followed the ubuntu versioning of the so file (0.0.0).

File: package/utils/lua5.3/Makefile (hunk 1 above)
Comment (rmilecki) [reply package/utils/lua5.3/Makefile]: @jow- just pointed out https://abi-laboratory.pro/index.php?view=timeline&l=lua to me
It seems Lua keeps ABI compatibility for all 5.3.x releases, so "ABI_VERSION should be 5.3"

Comment (cwluxul) [reply package/utils/lua5.3/Makefile]: @rmilecki  - updated ABI_VERSION to just 5.3 based on your request.

**Comment by rmilecki** [comment]
Thanks for your work on this @cwluxul!

I've merged this and while doing that I squashed those two commits. The later was needed due to changes from the first one, so it made sense to have that handled in a single patch.

**Pull request closed without merging**